examples/plot_incr_kmeans.py

```diff
@@ -0,0 +1,274 @@
+"""
+Incremental KMeans
+==================
+
+In an active learning setting, the trade-off between exploration and
+exploitation plays a central role. Exploration, or diversity, is
+usually enforced using coresets or, more simply, a KMeans algorithm.
+KMeans is therefore used to select samples that are spread across the
+dataset in each batch. However, as far as we know, maintaining diversity
+through the whole experiment is something that has not been considered.
+
+By allowing to start the optimization with fixed cluster centers, our
+Incremental KMeans allows for an optimal exploration of the space since
+samples are less likely to be selected nearby an already selected sample.
+
+This notebook introduces the principle behind Incremental KMeans and runs
+a small benchmark on generated data.
+
+"""
+
+from sklearn.cluster import MiniBatchKMeans
+from sklearn.metrics import pairwise_distances_argmin_min
+from cardinal.kmeans import IncrementalMiniBatchKMeans
+from sklearn.datasets import make_blobs
+from matplotlib import pyplot as plt
+from scipy.stats import ttest_rel
+import numpy as np
+
+##############################################################################
+# Let us first define the parameters of our experiment. We will generate a
+# dataset from 8 clusters and we will try to keep 4 clusters fixed. We do
+# this example in 2 dimensions for visualization purposes.
+
+n_clusters = 8
+n_fixed_clusters = 4
+
+X, y, centers = make_blobs(centers=n_clusters, return_centers=True,
+                           n_features=2, random_state=2)
+
+##############################################################################
+# We define plotting functions. This function allows to plot the data colored
+# by clusters, and allows to displays fixed cluster centers in red and regular
+# centers in blue. We use it to display the ground truth.
+
+def plot_clustering(label, y_pred, centers, fixed_centers=[], inertia=None):
+    colors = ['C{}'.format(i) for i in y_pred]
+    plt.scatter(X[:, 0], X[:, 1], c=colors)
+    plt.scatter(centers[:, 0], centers[:, 1], c='blue', edgecolors='k',
+                marker='P', s=200)
+    if len(fixed_centers) > 0:
+        plt.scatter(fixed_centers[:, 0], fixed_centers[:, 1],
+                    c='red', edgecolors='k', marker='P', s=200)
+    plt.xticks([])
+    plt.yticks([])
+    plt.tight_layout(rect=[0, 0.03, 1, 0.95])
+    if inertia:
+        label = label + ", intertia={}".format(inertia)
```

Comment:
Maybe we don't need all that decimals :) 
```suggestion
        label = label + ", intertia={0:0.2f}".format(inertia)
```